doc/CONFIGURATION.md

```diff
@@ -444,7 +453,11 @@ Configuration Options:
 
 ### DHT Options
 
-The `dht_options` section is used to provide optional control of kad dht we instantiate. It currently only supports one setting `persistentDHT` which is used to store DHT records on disk. It enables persisting DHT records long-term, and avoiding storing them in memory which has the side-effect of reducing memory consumption. If set to true, it uses datastore key [namespaces](https://github.com/ipfs/go-datastore/tree/master/namespace) to wrap around the main storage layer of our node. This means that it will use whatever datastore type you define in `node.storage`, and wrap all keys with a `dhtdatastore` prefix.
+The `dht_options` section is used to provide optional control of kad dht we instantiate. It currently supports one setting `persistentDHT` which is used to store DHT records on disk. It enables persisting DHT records long-term, and avoiding storing them in memory which has the side-effect of reducing memory consumption. If set to true, it uses datastore key [namespaces](https://github.com/ipfs/go-datastore/tree/master/namespace) to wrap around the main storage layer of our node. This means that it will use whatever datastore type you define in `node.storage`, and wrap all keys with a `dhtdatastore` prefix.
+
+### Host Options
+
+The `host_options` section is used to provide optional control of libp2p host configurations. It currently supports one setting `natPortMap` which is used to enabled nat port mapping capabilities, and can be useful in situations where punching through NAT is needed.
```

Comment:
More explaining is needed for how this NAT transversal works. UPNP or some other method. Also how to test and debug.

Reply:
I added a link to the `go-libp2p-nat` library which is what powers the nat port mapping stuff.

Reply:
That readme also leaves much to be desired. From source, it's using NAT-PMP and UPNP.

Reply:
It's certainly not desirable but tbh I've rewritten more ipfs code than I want, and the last thing I want to do is struggle to read through some half assed documentation for the thousandth time lmao